prometheus-operator.yaml

```
@@ -137,26 +137,159 @@ update:
     - name: "Setup mock Kubernetes cluster"
       uses: test/kwok/cluster
-    - name: "Test prometheus-operator with mock API server"
+    # Instead of using the full bundle.yaml which contains large OpenAPI validation schemas in
+    # annotations, we have created minimal CRDs that only include the essential fields needed for testing
+    - name: "Setup minimal CRDs"
       runs: |
-        output=$(operator --log-level=debug \
+        kubectl create namespace prometheus-operator-test
+
+        cat <<EOF | kubectl apply -f -
+        apiVersion: apiextensions.k8s.io/v1
+        kind: CustomResourceDefinition
+        metadata:
+          name: prometheuses.monitoring.coreos.com
+        spec:
+          group: monitoring.coreos.com
+          names:
+            kind: Prometheus
+            plural: prometheuses
+            singular: prometheus
+          scope: Namespaced
+          versions:
+          - name: v1
+            served: true
+            storage: true
+            schema:
+              openAPIV3Schema:
+                type: object
+                properties:
+                  spec:
+                    type: object
+                    properties:
+                      replicas:
+                        type: integer
+                      serviceAccountName:
+                        type: string
+                      serviceMonitorSelector:
+                        type: object
+                      resources:
+                        type: object
+                        properties:
+                          requests:
+                            type: object
+                            properties:
+                              memory:
+                                type: string
+        ---
+        apiVersion: apiextensions.k8s.io/v1
+        kind: CustomResourceDefinition
+        metadata:
+          name: servicemonitors.monitoring.coreos.com
+        spec:
+          group: monitoring.coreos.com
+          names:
+            kind: ServiceMonitor
+            plural: servicemonitors
+            singular: servicemonitor
+          scope: Namespaced
+          versions:
+          - name: v1
+            served: true
+            storage: true
+            schema:
+              openAPIV3Schema:
+                type: object
+                properties:
+                  spec:
+                    type: object
+                    properties:
+                      selector:
+                        type: object
+                        properties:
+                          matchLabels:
+                            type: object
+                            additionalProperties:
+                              type: string
+                      endpoints:
+                        type: array
+                        items:
+                          type: object
+                          properties:
+                            port:
+                              type: string
+        EOF
+
```

Let's apply stripped CRDs here:
```suggestion
        kubectl apply -f https://github.com/prometheus-operator/prometheus-operator/releases/download/v${{package.version}}/stripped-down-crds.yaml

```